roles/ranger/admin/tasks/config.yml

```
@@ -31,3 +31,12 @@
   template:
     src: ranger-admin-services.sh.j2
     dest: '{{ ranger_install_dir }}/ews/ranger-admin-services.sh'
+
+- import_role:
+    name: tosit.tdp.utils.xml
+    tasks_from: merge
+  vars:
+    source_file: '{{ ranger_install_dir }}/conf/ranger-admin-site.xml'
+    merge_var: '{{ ranger_admin_site  }}'
+    dest_file: '{{ ranger_install_dir }}/conf/ranger-admin-site.xml'
+    with_xml_description: true
```

The Ansible `xml` module is enough to replace a few properties in the file. Here I just replace the `value` tags:
```suggestion
- name: Overwrite configurations not set by setup.sh
  xml:
    path: "{{ ranger_install_dir }}/conf/ranger-admin-site.xml"
    xpath: "/configuration/property[name='{{ item.key }}']/value"
    value: "{{ item.value }}"
    pretty_print: yes
  loop: "{{ ranger_admin_site | dict2items }}"
```

---

@Nuttymoon thank you very much for your help :)
I tried this module, and the problem is that the task fails when the property doesn't exist.

---

Indeed (although I think that they exist by default in `ranger-admin-site.xml`). But you can check if properties exist and create them if not:
```yaml
- name: Check if properties to overwrite exist
  xml:
    path: "{{ ranger_install_dir }}/conf/ranger-admin-site.xml"
    xpath: "/configuration/property[name='{{ item }}']"
  loop: "{{ ranger_admin_site.keys() | list }}"
  failed_when: no
  register: setup_sh_overwrite_props

- name: Create missing properties to overwrite
  xml:
    path: "{{ ranger_install_dir }}/conf/ranger-admin-site.xml"
    xpath: /configuration
    add_children:
      - property:
          _:
            - name: "{{ item.item }}"
            - value: "{{ ([item.item] | map('extract', ranger_admin_site) | list)[0] }}"
    pretty_print: yes
  loop: "{{ setup_sh_overwrite_props.results }}"
  when: '"Can''t process Xpath" in (item.msg | default(""))'
  register: setup_sh_overwrite_props
```

---

It's a bit complicated but still better than introducing a new module IMO. Moreover this is temporary until we rewrite the `setup.sh`.

---

thanks @Nuttymoon, indeed, i did something like this :) that's perfect, thank you again !

---

I simplified the snippet up there!

---

perfect :) Many thanks @Nuttymoon !!